### ISSUE TITLE
Move all board ports to cleaner and more precise `drivers/periph` model

### ISSUE THREAD

**Issue by authmillenon** [RIOT-OS/RIOT#1540]
- [x] arduino-due
- [ ] avsextrem (PWM implemented)
- [ ] chronos
- [x] mbed_lpc1768
- [ ] msb-430
- [ ] msb-430h
- [ ] msba2 (PWM implemented)
- [ ] native (cpuid implemented, GPIO: #1737)
- [ ] pttu (PWM implemented)
- [ ] qemu-i386
- [ ] redbee-econotag
- [x] stm32f0discovery
- [x] stm32f3discovery
- [x] stm32f4discovery
- [ ] telosb
- [x] udoo
- [ ] wsn430-v1_3b
- [ ] wsn430-v1_4
- [ ] z1

If you are going to fix one of these, please open a new issue, link to it here, assign yourself and go ahead and fix it. Then, assign @haukepetersen  as a reviewer. (Please don't open an issue unless you intend to fix it.)

**Comment by authmillenon** [comment]
Superseded by #3418
